test/docker/kafka/kafka_test.go

```diff
@@ -0,0 +1,217 @@
+	}
+	return &kafkaRuntime{topics: topics, Runtime: *br}, nil
+}
+
+func (k *kafkaRuntime) setup() error {
+
+	var err error
+
+	runOptions := &dockertest.RunOptions{Repository: "wurstmeister/zookeeper",
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			docker.Port(fmt.Sprintf("%s/tcp", zookeeperPort)): {{HostIP: "", HostPort: zookeeperPort}},
+			// port 22 is too generic to be used for the test
+			"29/tcp":   {{HostIP: "", HostPort: "22"}},
+			"2888/tcp": {{HostIP: "", HostPort: "2888"}},
+			"3888/tcp": {{HostIP: "", HostPort: "3888"}},
+		},
+	}
+	zookeeper, err := k.RunWithOptions(runOptions)
+	if err != nil {
+		return fmt.Errorf("could not start zookeeper: %w", err)
+	}
+
+	ip := zookeeper.Container.NetworkSettings.Networks["bridge"].IPAddress
+
+	kafkaTCPPort := fmt.Sprintf("%s/tcp", kafkaPort)
+
+	runOptions = &dockertest.RunOptions{
+		Repository: "wurstmeister/kafka",
+		Tag:        "2.12-2.5.0",
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			docker.Port(kafkaTCPPort): {{HostIP: "", HostPort: kafkaPort}},
+		},
+		ExposedPorts: []string{kafkaTCPPort},
+		Mounts:       []string{"/tmp/local-kafka:/etc/kafka"},
+		Env: []string{
+			"KAFKA_ADVERTISED_HOST_NAME=127.0.0.1",
+			fmt.Sprintf("KAFKA_CREATE_TOPICS=%s", strings.Join(k.topics, ",")),
+			fmt.Sprintf("KAFKA_ZOOKEEPER_CONNECT=%s:%s", ip, zookeeperPort),
+		}}
+
+	_, err = k.RunWithOptions(runOptions)
+	if err != nil {
+		return fmt.Errorf("could not start kafka: %w", err)
+	}
+
+	return k.Pool().Retry(func() error {
+		consumer, err := NewConsumer()
+		if err != nil {
+			return err
+		}
+		topics, err := consumer.Topics()
+		if err != nil {
+			log.Infof("err or during topic retrieval = %v", err)
+			return err
+		}
+
+		return validateTopics(topics, k.topics)
+	})
+}
+
+func validateTopics(clusterTopics, wantTopics []string) error {
+	var found int
+	for _, wantTopic := range wantTopics {
+		topic := strings.Split(wantTopic, ":")
+		for _, clusterTopic := range clusterTopics {
+			if topic[0] == clusterTopic {
+				found++
+			}
+		}
+	}
+
+	if found != len(wantTopics) {
+		return fmt.Errorf("failed to find topics %v in cluster topics %v", wantTopics, clusterTopics)
+	}
+
+	return nil
+}
+
+// NewProducer creates a new sync producer.
+func NewProducer() (sarama.SyncProducer, error) {
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+	config.Producer.Return.Errors = true
+
+	brokers := []string{fmt.Sprintf("%s:%s", kafkaHost, kafkaPort)}
+
+	return sarama.NewSyncProducer(brokers, config)
+}
+
+// NewConsumer creates a new consumer.
+func NewConsumer() (sarama.Consumer, error) {
+	config := sarama.NewConfig()
+	config.Consumer.Return.Errors = true
+
+	return sarama.NewConsumer(Brokers(), config)
+}
+
+// Brokers returns a list of brokers.
+func Brokers() []string {
+	return []string{fmt.Sprintf("%s:%s", kafkaHost, kafkaPort)}
+}
+
+func getTopic(name string) string {
+	return fmt.Sprintf("%s:1:1", name)
```

Comment:
should we maybe have a test also with more partitions / consumers ?

Reply:
Do you believe we should consume with two consumers for simple and consumer group and test the outcome?

Reply:
yes, it might be a useful scenario. Do you think it would be worth to have an issue for that ?

Reply:
We could in order to make sure that it works correctly... Care to open one?